### DISCUSSION TITLE
Ltex-ls LSP won't install

### DISCUSSION THREAD

**Discussion by jacopoaltieri** [mason-org/mason.nvim#1728] [Q&A] [answered]
Greetings. I have a neovim-lua configuration with mason as my LSP provider.  In my mason.lua file i correctly have 
```mason_lspconfig.setup({
			-- list of servers for mason to install
			ensure_installed = {
				"tsserver",
				"html",
				"cssls",
				"tailwindcss",
				"svelte",
				"lua_ls",
				"ltex",
				"graphql",
				"emmet_ls",
				"prismals",
				"pyright",
			},
		})```

However, when opening a .tex file i get the following error message: " ...ps/neovim/current/share/nvim/runtime/lua/vim/lsp/rpc.lua:800: Spawning language server with cmd: `{ "ltex-ls.bat" }` failed. The language server is either not installed, missing from PATH, or not executable."

If I run `:LspInstall` I am given the option to install ltex. I complete the installation, reopen vim, but the problem persists. Moreover, if I run once again the command, the same option is given to me.

I am pretty new to the neovim and mason world, so If I didn't provide some information needed please ask me for it and I will post everything relevant.
Thanks

**Comment by ilias777** [comment, marked as answer]
Do you add the ltex language server in nvim-lspconfig?

Make sure you have in your lsp config file this:

```lua
lspconfig.ltex.setup({
    -- ltex settings
})
```

Look here &rarr; [nvim-lspconfig quickstart](https://github.com/neovim/nvim-lspconfig?tab=readme-ov-file#quickstart)
and here &rarr; [server configuration for ltex](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#ltex)

And make sure you have `mason.nvim`, `mason-lspconfig.nvim` and `nvim-lspconfig` in this order in your lspconfig file:

1. `mason.nvim`: `require('mason').setup({ -- your settigns })`
2. `mason-lspconfig`: `require('mason-lspconfig').setup({ -- your settings })`
3.  `nvim-lspconfig`: `lspconfig.ltex.setup({ -- your settings })`

Look here &rarr; [mason-lspconfig setup](https://github.com/williamboman/mason-lspconfig.nvim#setup)
